### PR TITLE
Feature - IoT/Sql separate connection string tracking in diff libraries

### LINKED ISSUE #109: Provide capability to track a IoT Hub dependency based on connection string information
Provide capability to track a IoT Hub dependency based on connection string information.

Information can be resolved via [`IotHubConnectionStringBuilder.Create`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.devices.client.iothubconnectionstringbuilder.create?view=azure-dotnet#Microsoft_Azure_Devices_Client_IotHubConnectionStringBuilder_Create_System_String_).

### PR DESCRIPTION
**Pull request by stijnmoreels**
Closes #109

### CODE REVIEW

**Comment by stijnmoreels** [comment]
Also, before merging: I had a question [here](https://github.com/arcus-azure/arcus.observability/issues/109#issuecomment-653383792).

File: src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs [hunk 1]
```diff
@@ -557,43 +557,6 @@ public static void LogHttpDependency(this ILogger logger, HttpRequestMessage req
...
-        ///     Logs a SQL dependency
-        /// </summary>
-        /// <param name="logger">Logger to use</param>
-        /// <param name="connectionString">SQL connection string</param>
-        /// <param name="tableName">Name of table</param>
-        /// <param name="operationName">Name of the operation that was performed</param>
-        /// <param name="isSuccessful">Indication whether or not the operation was successful</param>
-        /// <param name="measurement">Measuring the latency to call the SQL dependency</param>
-        /// <param name="context">Context that provides more insights on the dependency that was measured</param>
-        public static void LogSqlDependency(this ILogger logger, string connectionString,  string tableName, string operationName, DependencyMeasurement measurement, bool isSuccessful, Dictionary<string, object> context = null)
```
Comment (tomkerkhove) [inline src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs:570]: Hold yo horses, we need to mark it as obsolete and create an issue to remove it for the next major release 😅

File: src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs [hunk 2]
```diff
@@ -557,43 +557,6 @@ public static void LogHttpDependency(this ILogger logger, HttpRequestMessage req
...
-        /// </summary>
-        /// <param name="logger">Logger to use</param>
-        /// <param name="connectionString">SQL connection string</param>
-        /// <param name="tableName">Name of table</param>
-        /// <param name="operationName">Name of the operation that was performed</param>
-        /// <param name="startTime">Point in time when the interaction with the HTTP dependency was started</param>
-        /// <param name="duration">Duration of the operation</param>
-        /// <param name="isSuccessful">Indication whether or not the operation was successful</param>
-        /// <param name="context">Context that provides more insights on the dependency that was measured</param>
-        public static void LogSqlDependency(this ILogger logger, string connectionString,  string tableName, string operationName, DateTimeOffset startTime, TimeSpan duration,  bool isSuccessful, Dictionary<string, object> context = null)
```
Comment (tomkerkhove) [inline src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs:589]: Hold yo horses, we need to mark it as obsolete and create an issue to remove it for the next major release 😅

File: src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs (hunk 1 above)
Comment (stijnmoreels) [reply src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs:570]: Yes, I was planning so, but it becomes ambigious that way. So not sure how to solve this just yet.

Comment (stijnmoreels) [reply src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs:570]: We need something with the same signature in one part obsolete and in another part not. Without changing names, arguments, ... 🤔 . So I removed it for now. Not sure, any ideas?

Comment (tomkerkhove) [reply src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs:570]: I'd keep it as is then and introduce a separate SQL package as part of v1.0; there's no way around it I'm afraid

Comment (tomkerkhove) [reply src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs:570]: Optional: Create issue to track this

Comment (stijnmoreels) [reply src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs:570]: #131

**Pull request merged**